src/Common/Enum/Enum.php

```diff
@@ -35,13 +35,26 @@ public function __construct($value);
      */
     public static function __callStatic(string $enum, array $args = []): Enum;
 
+    /**
+     * Return the enum's valid values
+     *
+     * @return mixed
+     */
+    public static function getValues(): array;
+
     /**
      * Return the enum value
      *
      * @return mixed
      */
     public function getValue();
 
+    /**
+     * @param Return TRUE is enum is matching $value
+     * @return bool
+     */
+    public function is($value): bool;
```

Comment:
I'd prefer a `equals (Enum $enum)` method so that we can use this as a Value Object